src/Microsoft.ML.Api/DataViewConstructionUtils.cs

```diff
@@ -92,198 +360,126 @@ protected DataViewBase(IHostEnvironment env, string name, InternalSchemaDefiniti
                 return new[] { GetRowCursor(predicate, rand) };
             }
 
-            public abstract class DataViewCursorBase : RootCursorBase, IRowCursor
+            public abstract class DataViewCursorBase : InputRowBase<TRow>, IRowCursor
```

Comment:
>DataViewCursorBase [](start = 34, length = 18)

all of this code is a clone of something, right?

Reply:
Ah yes let me add a comment about that.

---
In reply to: [219578467](https://github.com/dotnet/machinelearning/pull/973#discussion_r219578467) [](ancestors = 219578467)